lib/unifex/code_generator/base_types/list.ex

```diff
@@ -74,18 +81,104 @@ defmodule Unifex.CodeGenerator.BaseTypes.List do
     end
 
     @impl BaseType
-    def generate_initialization(name, _ctx) do
-      ~g<#{name} = NULL;>
+    def generate_destruction(name, ctx) do
+      ~g"""
+      if(#{name} != NULL) {
+        for(unsigned int i = 0; i < #{name}_length; i++) {
+          #{BaseType.generate_destruction(ctx.subtype, :"#{name}[i]", ctx.generator)}
+        }
+        unifex_free(#{name});
+      }
+      """
+    end
+  end
+
+  defmodule CNode do
+    @moduledoc false
+    use Unifex.CodeGenerator.BaseType
+    alias Unifex.CodeGenerator.BaseType
+    @impl BaseType
+    def generate_native_type(ctx) do
+      prefix = if ctx.mode == :const, do: "const ", else: ""
+
+      [
+        "#{prefix}#{BaseType.generate_native_type(ctx.subtype, ctx.generator)}*",
+        {"int", "_length"}
+      ]
+    end
```

Comment:
this is the same as for NIF, let's bring it one level up

Reply:
The difference is in signedness. For `CNode` we need `int` because functions from `ei.h` require it (e.g. `ei_get_type`). The same situation is with `generate_destruction`

Reply:
The argument types should be the same between nifs and cnodes if we want to write code compatible with both. If some functions in ei.h need different types, we should convert/cast them